[FIX] Suppress -Wunused-private-field warnings

matchy233 opened the pull request:
# Description

This PR (partly) fixes `-Wunused-private-field` warnings when compiling with `clang` on macOS.

There are two unused private fields:
1. `snoar_` in `OpenSwathOSWWriter.h`
2. `pos_min_` of `SplineInterpolatedPeaks::Navigator` in `SplineInterpolatedPeaks.h`

Both unused fields can be removed since they were not used anywhere in the code.

However, I feel that the first unused field should not be trivially removed since in `OpenSwathTSVWriter`, we utilized `sonar_` to determine whether we should write `var_sonar_<options>`  in the output or not:

```c++
if (sonar_)
{
  ofs << "\tvar_sonar_lag\tvar_sonar_shape\tvar_sonar_log_sn\tvar_sonar_log_diff\tvar_sonar_log_trend\tvar_sonar_rsq";
}
```
But in OSW writer, we append `VAR_SONAR<options>` to `sql_feature_ms2` regardless of `sonar_` provided or not. I'm curious whether we should add an `if` statement checking the value of `sonar_` like in TSV writer. (Hence `sonar_` is suppressed only using `[[maybe_unused]]`)


# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

# Advanced commands (admins / reviewer only):
- `/reformat` (experimental) applies the clang-format style changes as additional commit
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds

jpfeuffer commented:
It feels like you are right and an if-case in the writer would be better.

cbielow commented:
dumb question from my side: why is `sonar_` flagged as unused? It is clearly used in `OpenSwathTSVWriter.cpp` ...
Is this simply a false positive from Clang?

matchy233 commented:
@cbielow sorry for not making it clear. OSW writer and TSV writer both have `sonar_` as a private flag. It's only unused in OSW writer. I mentioned TSV writer for comparison. :)

cbielow commented:
ah, I see.
thanks for the heads up

matchy233 commented:
@jpfeuffer @timosachsenberg I've updated the PR with `if (sonar_)` branch included. Would be nice to have the latest changes reviewed!

jpfeuffer commented:
Looks good to me but maybe a swath person can have a look.

jcharkow reviewed (approved):
From the SWATH side these changes look good to me. However have not worked with SONAR data in the past.

jpfeuffer commented:
@matchy233 python bindings need to be updated still

timosachsenberg commented:
I can do a quick edit

matchy233 commented:
Is this available to merge? @timosachsenberg  @jpfeuffer I think timo has already made the changes to the Python binding

jpfeuffer commented:
Unfortunately I don't see it. It is not in this PR, where it should be.

matchy233 commented:
Hmm.. I probably have overwritten his commit to this PR by the force push
after I rebased my local branch to develop... I modified and made the
changes in the newest push.

Julianus Pfeuffer ***@***.***> 于2022年5月4日周三 17:25写道：

> Unfortunately I don't see it. It is not in this PR, where it should be.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenMS/OpenMS/pull/6046#issuecomment-1117050083>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHVUZSK4N7SMLBLTMCBOEYLVIIYBBANCNFSM5TWUDSWQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jpfeuffer commented:
Looks good. Will merge after CI.
